adblock/gen_adblock.sh

```diff
@@ -43,55 +42,54 @@ echo "Removing possible temporary files.."
-     echo "Processing domains file @ $2"
-     curl --progress-bar $2 >> $tempoutlist
-    elif [ "$1" == "whitelist-domains" ]; then
-     echo "Processing whitelist domains file @ $2"
-     curl --progress-bar $2 >> $tempwhitelistoutlist
-    else
-      echo "Unknown file type = $1"
-    fi
-  else
-    echo "Missing site URL on line $line"
-  fi
-done < "$sites"
-
-echo "Combining User Custom block host..."
-cat $blocklist >> $tempoutlist
-
-if [ -f $tempwhitelistoutlist ]; then
-  echo "Removing any downloaded whitelist items..."
-  awk 'NR==FNR{a[$0];next} !($0 in a) {print $NF}' $tempwhitelistoutlist $tempoutlist > $outlist
-  mv $outlist $tempoutlist
-fi
-
-echo "Filtering required domains from adblock list..."
-awk 'NR==FNR{a[$0];next} !($0 in a) {print $NF}' $allowhost $tempoutlist > $outlist
-
-echo "Filtering user requested domains from adblock list..."
-awk 'NR==FNR{a[$0];next} !($0 in a) {print $NF}' $permlist $tempoutlist > $outlist
-
-echo "Removing duplicate formatting from the domain list..."
-cat $outlist | sed -r -e 's/[[:space:]]+/\t/g' | sed -e 's/\t*#.*$//g' | sed -e 's/[^a-zA-Z0-9\.\_\t\-]//g' | sed -e 's/\t$//g' | sed -e '/^#/d' | sed -e '/^\([0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}\).*$/d' | sed -e 's/^[ \t]*//;s/[ \t]*$//' | sort -u | sed '/^$/d' | awk -v "IP=$destinationIP" '{sub(/\r$/,""); print IP" "$0}' > $finalist
+  for url in $(echo $line); do
+  echo "Attempting to Download $url"
+  curl --progress-bar $url | grep -v "#" | grep -v "::1" | grep -v "0.0.0.0 0.0.0.0" | sed '/^$/d' | sed 's/\ /\\ /g' | awk '{print $NF}' | grep -v '^\\' | grep -v '\\$'| sort >> $list
+  dos2unix $list
+  done
+  done < "$sites"
+}
+
+filter_file () {
+  filter=$1
+  original=$2
+  awk 'NR==FNR{a[$0];next} !($0 in a) {print $NF}' $filter $original | sort -u > ${original}.tmp
+  mv ${original}.tmp $original
+}
+
+cleanup () { 
+  unclean=$1
+  output=$2
+  cat $unclean | grep -oE "\b([0-9]{1,3}\.){3}[0-9]{1,3}\b" | sort -u > ${output}.tmp
+  mv ${output}.tmp $output
+  awk 'NR==FNR{a[$0];next} !($0 in a) {print $NF}' $output $unclean | sort -u > ${output}.tmp
+  mv ${output}.tmp $output
+  cat $output | sed -r -e 's/[[:space:]]+/\t/g' | sed -e 's/\t*#.*$//g' | sed -e 's/[^a-zA-Z0-9\.\_\t\-]//g' | sed -e 's/\t$//g' | sed -e '/^#/d' | sed -e 's/^[ \t]*//;s/[ \t]*$//' | sort -u | sed '/^$/d' > $3
+}
+
+download_file $blocksites $tempoutlist
+[ -f $allowsites ] && echo "Downloading User Allow List..." && download_file $allowsites $tempwhitelistoutlist
+[ -f $blocklist ] && echo "Combining User Custom block host..." && cat $blocklist >> $tempoutlist
+[ -f $tempwhitelistoutlist ] && echo "Removing any downloaded whitelist items..." && filter_file $tempwhitelistoutlist $tempoutlist
+[ -f $allowlist ] && echo "Filtering required domains from adblock list..." && filter_file $allowlist $tempoutlist
+[ -f $permlist ] && echo "Filtering user requested domains from adblock list..." && filter_file $permlist $tempoutlist
```

Comment:
This text is swapped.  perList is the required domain list, and allowlist is "user requested".